src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -0,0 +1,332 @@
+                        context.Results.Add(CreateApiDescription(routeEndpoint, httpMethod, methodInfo));
+                    }
+                }
+            }
+        }
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string httpMethod, MethodInfo methodInfo)
+        {
+            // Swashbuckle uses the "controller" name to group endpoints together.
+            // For now, put all methods defined the same declaring type together.
+            string controllerName;
+
+            if (methodInfo.DeclaringType is not null && !IsCompilerGenerated(methodInfo.DeclaringType))
+            {
+                controllerName = methodInfo.DeclaringType.Name;
+            }
+            else
+            {
+                // If the declaring type is null or compiler-generated (e.g. lambdas),
+                // group the methods under a "Map" controller.
+                controllerName = _environment.ApplicationName;
+            }
+
+            var apiDescription = new ApiDescription
+            {
+                HttpMethod = httpMethod,
+                RelativePath = routeEndpoint.RoutePattern.RawText?.TrimStart('/'),
+                ActionDescriptor = new ActionDescriptor
+                {
+                    RouteValues =
+                    {
+                        ["controller"] = controllerName,
+                    },
+                },
+            };
+
+            var hasJsonBody = false;
+
+            foreach (var parameter in methodInfo.GetParameters())
+            {
+                var parameterDescription = CreateApiParameterDescription(parameter, routeEndpoint.RoutePattern);
+
+                if (parameterDescription.Source == BindingSource.Body)
+                {
+                    hasJsonBody = true;
+                }
+
+                apiDescription.ParameterDescriptions.Add(parameterDescription);
+            }
+
+            AddSupportedRequestFormats(apiDescription.SupportedRequestFormats, hasJsonBody, routeEndpoint.Metadata);
+            AddSupportedResponseTypes(apiDescription.SupportedResponseTypes, methodInfo.ReturnType, routeEndpoint.Metadata);
+
+            return apiDescription;
+        }
+
+        private ApiParameterDescription CreateApiParameterDescription(ParameterInfo parameter, RoutePattern pattern)
+        {
+            var (source, name) = GetBindingSourceAndName(parameter, pattern);
+
+            return new ApiParameterDescription
+            {
+                Name = name,
+                ModelMetadata = CreateModelMetadata(parameter.ParameterType),
+                Source = source,
+                DefaultValue = parameter.DefaultValue,
+                Type = parameter.ParameterType,
+            };
+        }
+
+        // TODO: Share more of this logic with RequestDelegateFactory.CreateArgument(...) using RequestDelegateFactoryUtilities
```

Comment:
Could you file an issue and remove this?

Reply:
https://github.com/dotnet/aspnetcore/issues/33644